### ISSUE TITLE
"The file couldn't be opened" when attempting to change VM settings

### ISSUE THREAD

**Issue by emberdex** [utmapp/UTM#6639]
**Describe the issue**  

I noticed a few issues which might all be related:

- When creating new VMs, I get the "file couldn't be opened" error message, and the newly-created VM doesn't show up in the list until the app is restarted.
- When changing settings of any VM, the "file couldn't be opened" message appears when I click Save, and the settings do not take effect.
- Clicking the Install Windows Guest Tools menu option on a running VM does nothing.

I have tried granting UTM full disk access in macOS to see if this resolves the issue, to no avail.

![CleanShot 2024-08-29 at 21 09 50@2x](https://github.com/user-attachments/assets/ad184674-0e3c-4c88-87d6-9e131e16e283)

**Configuration**  

* UTM Version: 4.5.4
* macOS Version: macOS Sequoia Developer Beta 8 (build 24A5331b)
* Mac Chip (Intel, M1, ...): M1 (8-core GPU) - 8GB RAM

**Crash log**  

N/A (app isn't crashing)

**Debug log**  

N/A (I can't enable it due to the aforementioned issue)

**Upload VM**  

N/A (applies to all VMs)

**Comment by naveenrajm7** [comment]
Hello
Can you consistently reproduce the "The file couldn't be opened" issue? if so, can you please share the steps?

I had a slightly similar issue  https://github.com/utmapp/UTM/issues/6398 , which was fixed in the 4.5.4 release. 
(Also please check if you are using ```Version 4.5.4 (100)``` )

**Comment by Tangtangchannel** [comment]
> 我遇到了同样的问题。我使用的是 macOS Sequoia (24A5331b)。也许 UTM 应该针对 macOS Sequoia 进行优化和适配

I hardly encountered any similar issues with UTM before updating Sequoia (upgrading from an older Beta version to a newer Beta version), but I started experiencing this problem after upgrading to a higher Sequoia Beta version

**Comment by lassestilvang** [comment]
I have the same issue on macOS Sequoia RC (24A335).

Installed UTM for the first time (with Homebrew), created a couple of macOS and Ubuntu VM's and everything was working flawlessly. Then all of a sudden I started getting the "The file couldn't be opened" when creating a new VM and even editing an existing. Not sure what triggered it.

> It seems that by cleaning the ~/Library/Containers/com.utmapp.UTM.* folder, I was able to get the program running normally.

Unfortunately this didn't work for me.

I also tried uninstalling (`brew uninstall --cask utm --zap`) and installing again but still getting the error...

**Comment by mbert** [comment]
Same issue here. Cannot assign a boot ISO image for installing an OS. Installed Sequoia yesterday and running UTM 4.5.4 (100).

**Comment by ddv12138** [comment]
Same issue here.

**Comment by dciancu** [comment]
Same issue macOS Sequoia 15.0 (24A335).

**Comment by marcmento** [comment]
Also getting the same issue, macOS 15.0 (24A335). Can't edit VM without getting "file couldn't be opened" message.

**Comment by frostydev99** [comment]
Has anyone solved this?

**Comment by robifaustmann** [comment]
I have the same issue on macOS Sequoia 15.0 and UTM 4.4.5. I keep getting the error message but the changes seem to be saved. Changing the network settings on the VM gives an error message but **the changes are still saved**. The error message is not affecting the function and the **bug seems to be only cosmetical**.

Neither updating to UTM 4.5.4 nor buying the app from the App Store helps. The update 4.5.4 doesn't remove the error message.

**Comment by dcvr2469** [comment]
Same issue with macOS Sequoia 15.0 and UTM 4.5.4. Uninstalling/reinstalling UTM = no change. Clearing the mentioned folder(s), ~/Library/Containers/com.utmapp.UTM = no change.  Another problem, if it hasn't already been mentioned, immediately following the creation of a new VM, the VM does not show until closing and reopening UTM.

**Comment by scramlo** [comment]
Same issue here.

**Comment by cusspvz** [comment]
Same, related with the disk/iso path. You can also replicate by adding an ISO removable drive. Happens on both QEMU and Apple Virtualization.

<img width="808" alt="image" src="https://github.com/user-attachments/assets/6f3143d7-244e-4261-892a-83141fa15fcd">

<img width="808" alt="image" src="https://github.com/user-attachments/assets/a4223955-febd-4d14-b33c-bb50eaeb00f6">

**Comment by rbgarga** [comment]
Same issue here using sequoia

**Comment by G4G4N** [comment]
Same issue here, from when I updated to Sequoia

**Comment by ITmaze** [comment]
It appears that #6709 is another symptom.

I wonder if the issue is a permissions one caused by Sequoia. I note errors in launchd.log which might be related:

```txt
./com.apple.xpc.launchd/launchd.log:2024-10-03 13:26:33.882914 (pid/66653 [UTM]) <Warning>: failed activation: name = com.utmapp.UTM.apple-extension-service, flags = 0x0, requestor = com.apple.appki[71850], error = 1: Operation not permitted
./com.apple.xpc.launchd/launchd.log:2024-10-03 13:38:46.837092 (pid/72135 [UTM]) <Warning>: failed activation: name = com.utmapp.UTM.apple-extension-service, flags = 0x0, requestor = com.apple.appki[72139], error = 1: Operation not permitted
./com.apple.xpc.launchd/launchd.log:2024-10-03 13:40:09.131939 (pid/72158 [UTM]) <Warning>: failed activation: name = com.utmapp.UTM.apple-extension-service, flags = 0x0, requestor = com.apple.appki[72163], error = 1: Operation not permitted
./com.apple.xpc.launchd/launchd.log:2024-10-03 13:54:19.242187 (pid/73688 [UTM]) <Warning>: failed activation: name = com.utmapp.UTM.apple-extension-service, flags = 0x0, requestor = com.apple.appki[73701], error = 1: Operation not permitted
./com.apple.xpc.launchd/launchd.log:2024-10-03 13:56:19.552343 (pid/73734 [UTM]) <Warning>: failed activation: name = com.utmapp.UTM.apple-extension-service, flags = 0x0, requestor = com.apple.appki[73740], error = 1: Operation not permitted
./com.apple.xpc.launchd/launchd.log:2024-10-03 13:57:38.779096 (pid/73756 [UTM]) <Warning>: failed activation: name = com.utmapp.UTM.apple-extension-service, flags = 0x0, requestor = com.apple.appki[73765], error = 1: Operation not permitted
```

I note that I attempted to move machines around and the result was that the "moved" machine became unusable.

I also created two machines with the same name, neither opened the preferences pane when I ticked the box during creation.

I was unable to point at a .ISO file in my home directory and I was unable to set the shared directory to /Users/Shared which is what I'd been using before upgrading to Sequoia.

I was unable to locate any UTM specific logging and I'm fairly sure that the issue occurs outside qemu, as-in, this is a UTM issue. Existing machines that have not been edited appear to work normally.

I also note that for a period after upgrading to Sequoia I was able to use UTM normally. It wasn't until today that it all broke. I have only rebooted after the upgrade, six days ago.

I am unable to determine if there is a "helper" process that is running asynchronously that might be causing this.

Installed via homebrew:

```text
$ brew info utm 
==> utm: 4.5.4
https://mac.getutm.app/
Installed
/opt/homebrew/Caskroom/utm/4.5.4 (940.9MB)
  Installed using the formulae.brew.sh API on 2024-09-16 at 15:13:42
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/u/utm.rb
==> Name
UTM
==> Description
Virtual machines UI using QEMU
==> Artifacts
UTM.app (App)
/Applications/UTM.app/Contents/MacOS/utmctl (Binary)
==> Analytics
install: 3,389 (30 days), 9,243 (90 days), 35,596 (365 days)
```

**Comment by ITmaze** [comment]
Also #6707

**Comment by ITmaze** [comment]
Also #6699

**Comment by ITmaze** [comment]
Also #6692

**Comment by ITmaze** [comment]
I have just installed 15.0.1. This comes with a reboot.

I don't know if the reboot or the upgrade fixed this issue, but so far I can edit the preferences of a VM.

**Comment by osy** [comment]
Can someone who is experiencing this issue generate a [sysdiagnose](https://support.kandji.io/support/solutions/articles/72000599404-how-to-generate-a-sysdiagnose-in-macos) as soon as they experience the error (so it can be the last event that is captured)? Note that this will contain identifiers from your machine including username, location, etc so only share it if you feel comfortable. You can email dev (at) getutm (dot) app with the dump.

**Comment by chunfengyao** [comment]
Same issue, fixed after upgrading the system to `15.0.1 (24A348)`. It might caused by macOS itself.
also https://github.com/utmapp/UTM/issues/6699

**Comment by chunfengyao** [comment]
![QQ_1728027107872](https://github.com/user-attachments/assets/aa60c9fb-e645-47d4-a87a-dabec2f0def5)
![QQ_1728027254879](https://github.com/user-attachments/assets/66b82500-8f1c-4d37-a2d7-890b70a16c27)

**Comment by osy** [comment]
Can others in this thread confirm if 15.0.1 fixes it?

**Comment by rbgarga** [comment]
Same here, 15.0.1 fixed it

**Comment by rmatinata** [comment]
I will advise a bit of caution here... There are multiple symptoms captured across different bugs that seem to be related to the tightened Sequoia security.

I've experienced all of those:
- All my VMs not in the Container folder are suddenly grayed out 
- Can't import VMs
- Issues trying to create new VMs

What I did (and thought to be the solution) was:

- Run 'tccutil reset All com.utmapp.UTM'
- Reset

But then, a week later, all problems are back. So an update to 15.0.1 might give you the illusion is fixed (since it entails a reboot)

**Comment by freebrowser1** [comment]
Weird.
In my Macbook M1 Pro the Sequoia upgrade didn't affect UTM adversely. It runs Win11(x64) and Ubuntu(arm and x64) VM's without issue.

**Comment by ITmaze** [comment]
On my machine it worked fine on 15.0 for several days, until it didn't.

As I said in my comment, the fix might be the reboot, or it might be the
update to 15.0.1.

My machine, a MacBook Air, sleeps when not in use, so I'll be monitoring
this and attempt to get a log if it happens again.

--
finger painting on glass is an inexact art - apologies for any errors in
this scra^Hibble

()/)/)() ..ASCII for Onno..

**Comment by osy** [comment]
Reminder: https://github.com/utmapp/UTM/issues/6639#issuecomment-2392725949 if people are still experiencing the issue

**Comment by dcvr2469** [comment]
> Same issue with macOS Sequoia 15.0 and UTM 4.5.4. Uninstalling/reinstalling UTM = no change. Clearing the mentioned folder(s), ~/Library/Containers/com.utmapp.UTM = no change. Another problem, if it hasn't already been mentioned, immediately following the creation of a new VM, the VM does not show until closing and reopening UTM.

Upgrade to 15.0.1 seems to have resolved the issue for me

**Comment by ITmaze** [comment]
I'm sorry to report that the issue has now reoccurred after upgrading to macOS Sequoia 15.0.1. I have created a `sysdiagnose`. I actually created two. I saw the error. Quit UTM, created a sysdiagnose, launched UTM, generated the error, then created another sysdiagnose.

I am unable to determine which data in the 1,591 files that process generated contains personal and private information, so I'm uncomfortable with sending the entire archive.

Which files would you like to look at @osy ?

**Comment by osy** [comment]
To start with `system_logs.logarchive` is good

**Comment by ITmaze** [comment]
That directory alone is 373 MB and contains 717 files. Care to narrow it down a tad?

**Comment by osy** [comment]
You can grep for UTM (case sensitive) and only export lines containing that

**Comment by ITmaze** [comment]
Only three of the matching files are text files. This is the list of files that contain the string UTM. In total, compressed, they are 52M.

```
476K    ./Signpost/00000000000000f2.tracev3
4.0K    ./57/1EFCB0E9E53ED2A5ACCDD477CB16DE
4.0K    ./ED/C6578BAD13354383FA9725A19826CC
320K    ./7B/E6485FBD5B3702B50D63F6EFCB7E10
1.2M    ./2A/D9786F321D3C18A44E31BE04695BD7
 10M    ./logdata.LiveData.tracev3
1.4M    ./Special/00000000000004d9.tracev3
5.0M    ./Extra/logdata.statistics.1.jsonl
2.1M    ./Extra/logdata.statistics.0.jsonl
1.0M    ./Extra/logdata.statistics.1.txt
 10M    ./Persist/0000000000000629.tracev3
9.6M    ./Persist/0000000000000632.tracev3
121M    ./dsc/A02DCFBA3BDA3CA68A4212F00A4D75D3
```

**Comment by osy** [comment]
I guess they are compressed. Open the logarchive and it should open up Console. Type in UTM as the filter and see if you can export the resulting entries.

**Comment by ITmaze** [comment]
To create this I did the following:

* `sudo sysdiagnose -f ~/Desktop/`
* `cd ~/Desktop/sysdiagnose_2024.10.08_10-42-38+0800_macOS_Mac15-12_24A348`
* `open system_logs.logarchive`
* Wait for `Console` to launch
* Click "All Messages"
* Type `UTM` in the Search box and hit Enter.
* Select All
* Copy
* Open `TextEdit`
* Format -> Make Plain Text
* Paste
* Save As `utm.log`

[utm.log](https://github.com/user-attachments/files/17288410/utm.log)

**Comment by osy** [comment]
To be clear, you've gotten the error when trying to save VM settings? I don't see any references to config.plist in the logs. Maybe you can search for that in the logarchive?

**Comment by ITmaze** [comment]
```
error	2024-10-08 10:42:01.003515 +0800	kernel	System Policy: sysdiagnosed(13980) deny(1) file-read-xattr /Users/onno/Library/Group Containers/group.com.apple.PegasusConfiguration/config.plist
error	2024-10-08 10:42:01.003997 +0800	kernel	System Policy: sysdiagnosed(13980) deny(1) file-read-data /Users/onno/Library/Group Containers/group.com.apple.PegasusConfiguration/config.plist
error	2024-10-08 10:44:16.716873 +0800	kernel	System Policy: sysdiagnosed(13980) deny(1) file-read-xattr /Users/onno/Library/Group Containers/group.com.apple.PegasusConfiguration/config.plist
error	2024-10-08 10:44:16.717206 +0800	kernel	System Policy: sysdiagnosed(13980) deny(1) file-read-data /Users/onno/Library/Group Containers/group.com.apple.PegasusConfiguration/config.plist
```

**Comment by ITmaze** [comment]
In answer to your question, yes, I was trying to save a VM preference.

**Comment by ITmaze** [comment]
Also, since you're likely to ask, this is the output from `ls -l`:

```
-rw-r--r--@ 1 onno  staff  51424  8 Oct 13:53 /Users/onno/Library/Group Containers/group.com.apple.PegasusConfiguration/config.plist
```

And this is what `stat` says:

```
16777234 4123923 -rw-r--r-- 1 onno staff 0 51424 "Oct  8 13:58:05 2024" "Oct  8 13:53:40 2024" "Oct  8 13:53:40 2024" "Oct  8 13:53:40 2024" 4096 104 0 /Users/onno/Library/Group Containers/group.com.apple.PegasusConfiguration/config.plist
```

**Comment by osy** [comment]
Okay but that's unrelated to UTM. It should be an error accessing some file in your .utm bundle.

**Comment by ITmaze** [comment]
What is the name of the file that the preferences save uses?

**Comment by osy** [comment]
Example: `/Users/osy/Library/Containers/com.utmapp.UTM/Data/Documents/Linux.utm/config.plist`

**Comment by ITmaze** [comment]
I've just created a VM from scratch, quit UTM, since you cannot click on the machine, relaunched, attempted to update the preferences (by turning off UEFI) and clicking "Save".

I then created a new sysdiagnose, extracted the logs, opened it in Console, searched for a different config.plist, it's not there, just the one I already showed you.

**Comment by JoshHrach** [comment]
I'm on 15.0.1 and am seeing this error when trying to save changes to my Shared Folder (well, it seems to have forgotten my shared folder, and now I can't save it, so no sharing is possible... argh!)

I'll send out a sysdiagnose later today.

**Comment by rbgarga** [comment]
> Same here, 15.0.1 fixed it

Today, without upgrading neither macOS or UTM, the problem is back

**Comment by rbgarga** [comment]
Here you can find logs from sysdiagnose [UTM.log](https://github.com/user-attachments/files/17466428/UTM.log)

**Comment by tonilampela** [comment]
I don't know if this is at all related, but other apps are also seeing problems accessing files outside of macOS sandbox in Sequoia and apparently fix is incoming in macOS 15.1 which is due next week:

https://www.reddit.com/r/strongbox/comments/1fo7z20/comment/lp5mtnu/

**Comment by 4WvgqJu** [comment]
I've issues too with Sequoia.

**Comment by JoshHrach** [comment]
Just installed macOS 15.1 and the issue is resolved with it.

**Comment by osy** [comment]
Please update to 15.1 if you haven't already. Apple claims to have fixed the issue.

**Comment by jpwynn** [comment]
For Sonoma, is this the same "sandbox" issue discussed here for **Sequoia**? Or should I report this in a new post?

I am having precisely the same "The file couldn’t be opened" issue on **Sonoma** 14.7.1 (MBP M2 32GB) when trying to change _any_ settings or set my Shared Directory.

* Originally installed 4.5.4 under **Ventura**, with windows 11 arm, worked great.
* Upgraded to **Sonoma** 14.7.1 a several days ago
* Tried to open my windows 11arm VM but got the "Cannot access TPM data" error
* Upgraded from UTM 4.5.4 to 4.6.3 to see if that helped; no change
* First, I set permissions on the vm .utm file in the ~/Library/Containers/UTM/Data/Documents folder to give all users read_write permission (per another comment here on UTM issues)
* Next, per another comment thread I turned off Settings > QEMU > "TMP 2.0 Device" ... when I clicked 'save' I got the "The file couldn’t be opened" error HOWEVER that setting does stay off
* * Now I can run the VM, BUT... without the previously set shared directory
* The shared directory, which worked previously, is no longer present inside the Windows11arm VM, and if I quit the vm and re-try to set the shared directory, it gives the "The file couldn’t be opened" error message.

Interestingly, even though it will not let me "set" the shared directory, when I click the "browse" button to do so, it does default to whatever folder I last tried to set... so the selected folder does get "saved in there somewhere" but cannot be "set" where UTM uses it.

TLDR; changing any setting (for my preexisting W11 vm, and for that matter, also for my preexisting W10 vm) gives a "The file couldn’t be opened" error which prevents me form setting a Shared Directory

I suppose it's nice to know that _IF_ it is the same sandbox issue as reported here on Sequoia the problem should go away if I bite the bullet and upgrade to Sequoia, but since this is my primary Dev't machine there are ample reasons to think I ought to wait several more months until a .2 or .3 release (as I usually do given the sad state of Apple's QA these days)... for example the numerous timemachine, SMB, spotlight, photos, window placement, VPN, keychain/secure notes, and other ongoing issues reported in reddit.com/r/MacOS/comments/1fihlge/macos_15_sequoia_bugs_and_issues_megathread

**Comment by ITmaze** [comment]
While I'm not the developer, I think it's appropriate to note here that
this is being experienced in both versions of the OS.

Although I cannot speak directly to the question about Sonoma, I note that
when I reported the original bug, there appeared to be some sort of
relationship between the bug occuring and the Mac having been rebooted.

I was not able to discover the specifics of this relationship, other than
to say that I observed that immediately after reboot it all appeared to
work fine, but sometime later it didn't.

I wonder if this might also be true for Sonoma.

**Comment by jpwynn** [comment]
> I was not able to discover the specifics of this relationship, other than to say that I observed that immediately after reboot it all appeared to work fine, but sometime later it didn't. I wonder if this might also be true for Sonoma.

Thanks. AFAIK I have only one other app that after "upgrading" from Ventura to Sonoma has inexplicable permissions issues during configuration (Cookie.app a clipboard manager) so it might be related.

I find the UTM issue gives the "The file couldn’t be opened" 100% of the time even immediately after a reboot.

**Comment by GeoffatMM** [comment]
I am also getting this message but if I cancel out of the window then go back into the VM settings, the change has taken place.

**Comment by mortea** [comment]
I'm on Sequoia 15.3.1 and running UTM 4.6.4 newly installed by homebrew. I still get the error whenever I try to save a setting. I've tried cancelling and it does not save the setting.

**Comment by 0xb0mb3r** [comment]
Im on macos 15.0 and upgrading to UTM Version [v4.6.4](https://github.com/utmapp/UTM/releases/tag/v4.6.4) fixed it for me

**Comment by csgordon** [comment]
I'm on macos 15.0 and v4.7.5 still has this bug for me
